Clarify ADSync service account password change process (#1954)
MicrosoftDocs/entra-docs: docs/identity/hybrid/connect/how-to-connect-sync-change-serviceacct-pass.md

ms.custom: sfi-ga-nochange, sfi-image-nochange
---
# Changing the ADSync service account password
Changing the ADSync service account password can prevent the Synchronization Service from starting successfully. When this occurs, the encryption key must be discarded and re-created, and the passwords for both the AD Connector account and the Microsoft Entra ID Connector account must be reconfigured. 

> >[!IMPORTANT]

Reviewer: **[Pre-existing] BLOCKING — Admonition spacing**: Missing space after `>`. Must be `> [!IMPORTANT]` not `>[!IMPORTANT]`.

```suggestion
> [!IMPORTANT]
```

> If you use Connect with a build from 2017 March or earlier, then you should not reset the password on the service account since Windows destroys the encryption keys for security reasons. You can't change the account to any other account without reinstalling Microsoft Entra Connect. If you upgrade to a build from 2017 April or later, then it is supported to change the password on the service account, but you can't change the account used. 

Microsoft Entra Connect, as part of the Synchronization Service, uses an encryption key to store the passwords of the AD DS Connector account and Microsoft Entra ID Connector account. These accounts are encrypted before they're stored in the database. 

The encryption key used is secured using [Windows Data Protection (DPAPI)](/previous-versions/ms995355(v=msdn.10)). DPAPI protects the encryption key using the **ADSync service account**. 

> If you need to change the ADSync service account password you can use the procedures in [Abandoning the ADSync service account encryption key](#abandoning-the-adsync-service-account-encryption-key) to accomplish this.  These procedures should also be used if you need to abandon the encryption key for any reason.

Reviewer: **Suggestion — Double space + missing comma**: Two spaces after "this." and missing comma after the introductory clause "If you need to change the ADSync service account password".

```suggestion
If you need to change the ADSync service account password, you can use the procedures in [Abandoning the ADSync service account encryption key](#abandoning-the-adsync-service-account-encryption-key) to accomplish this. These procedures should also be used if you need to abandon the encryption key for any reason.
```

If you need to abandon the encryption key, use the following procedures to accomplish that.

1. [Stop the Synchronization Service](#1-stop-the-synchronization-service)

> 2. [Abandon the existing encryption key](#2-abandon-the-existing-encryption-key)

Reviewer: **BLOCKING — Numbering format**: Use `1.` for all ordered list items (markdown auto-numbering), not sequential `1.`, `2.`, `3.`, etc. This applies to lines 50–58.

```suggestion
1. [Abandon the existing encryption key](#2-abandon-the-existing-encryption-key)
```

3. [Start the Synchronization Service](#3-start-the-synchronization-service)
   
4. [Provide the password of the AD DS Connector account](#4-provide-the-password-of-the-ad-ds-connector-account)

> 5. [Reinitialize the password of the Entra ID Connector account](#5-reinitialize-the-password-of-the-entra-id-connector-account)

Reviewer: **BLOCKING — Product naming**: Always use "Microsoft Entra" — never "Entra" alone. This appears in the link text and the heading anchor.

```suggestion
1. [Reinitialize the password of the Microsoft Entra ID Connector account](#5-reinitialize-the-password-of-the-microsoft-entra-id-connector-account)
```

  
  
#### 1. Stop the Synchronization Service

> First you can stop the service in the Windows Service Control Manager.  Make sure that the service isn't running when attempting to stop it.  If it is, wait until it completes and then stop it.

Reviewer: **Suggestion — Double spaces**: Two spaces after "Manager." and after "it."

```suggestion
First you can stop the service in the Windows Service Control Manager. Make sure that the service isn't running when attempting to stop it. If it is, wait until it completes and then stop it.
```

> 1.1. Go to Windows Service Control Manager (START → Services).

Reviewer: **BLOCKING — Sub-step numbering format**: `1.1.`, `1.2.` etc. aren't valid markdown list syntax — they render as plain text paragraphs, not as numbered lists. Use standard numbered lists (`1.`) with 4-space indentation under the parent step. This issue affects all sub-steps throughout the article (lines 66–67, 74–80, 90–91, 98–105, 114–116).

```suggestion
1. Go to Windows Service Control Manager (START → Services).
```

> 1.2. Select **Microsoft Entra ID Sync** and click Stop.

Reviewer: **BLOCKING — Input verb + missing bold**: Use "select" instead of "click" for UI actions. The UI element "Stop" must be bold.

```suggestion
1. Select **Microsoft Entra ID Sync** and select **Stop**.
```

  
  
#### 2. Abandon the existing encryption key
Abandon the existing encryption key so that new encryption key can be created:

2.1. Sign in to your Microsoft Entra Connect Server as administrator.

2.2. Start a new PowerShell session.

2.3. Navigate to folder: `'$env:ProgramFiles\Microsoft Azure AD Sync\bin\'`

2.4. Run the command: `./miiskmu.exe /a`

![Screenshot that shows PowerShell after running the command.](./media/how-to-connect-sync-change-serviceacct-pass/key5.png)

  
  
#### 3. Start the Synchronization Service

> Now that the Synchronization Service has access to the encryption key and all the passwords it needs, you can restart the service in the Windows Service Control Manager:

Reviewer: **BLOCKING — Incorrect description after reordering**: This text says the service "has access to the encryption key and all the passwords it needs" but at step 3, passwords haven't been reconfigured yet (that's steps 4 and 5). The description was carried over from the original step 5 and needs updating for the new order.

```suggestion
After abandoning the old encryption key, restart the Synchronization Service so that it can generate a new encryption key. You can then reconfigure the required account passwords in the following steps:
```

3.1. Go to Windows Service Control Manager (START → Services).

> 3.2. Select **Microsoft Entra ID Sync** and click Restart.

Reviewer: **BLOCKING — Input verb + missing bold**: Use "select" instead of "click". "Restart" must be bold.

```suggestion
1. Select **Microsoft Entra ID Sync** and select **Restart**.
```

  
  
#### 4. Provide the password of the AD DS Connector account
As the existing passwords stored inside the database can no longer be decrypted, you need to provide the Synchronization Service with the password of the AD DS Connector account. The Synchronization Service encrypts the passwords using the new encryption key:

4.1. Start the Synchronization Service Manager (START → Synchronization Service).

> </br>![Sync Service Manager](./media/how-to-connect-sync-change-serviceacct-pass/startmenu.png)  

Reviewer: **BLOCKING — Alt text accessibility (3 violations)**:
1. **Missing prefix**: Alt text must start with "Screenshot of..." or similar descriptive type.
2. **Missing period**: Alt text must end with a period.
3. **Too short**: "Sync Service Manager" is ~20 characters (minimum 40).

Also, `</br>` is invalid HTML — use `<br>`.

```suggestion
<br>![Screenshot of the Synchronization Service Manager start menu shortcut.](./media/how-to-connect-sync-change-serviceacct-pass/startmenu.png)
```

4.2. Go to the **Connectors** tab.
4.3. Select the **AD Connector** that corresponds to your on-premises AD. If you have more than one AD connector, repeat the following steps for each of them.
4.4. Under **Actions**, select **Properties**.

> 4.5. In the pop-up dialog, select **Connect to Active Directory Forest**:

Reviewer: **BLOCKING — UI punctuation**: Drop trailing colon from the UI element label per style guide.

```suggestion
1. In the pop-up dialog, select **Connect to Active Directory Forest**.
```

4.6. Enter the password of the AD DS connector account in the **Password** textbox. If you don't know its password, you must set it to a known value before performing this step.

> 4.7. Click **OK** to save the new password and close the pop-up dialog.

Reviewer: **BLOCKING — Input verb**: Use "Select" instead of "Click".

```suggestion
1. Select **OK** to save the new password and close the pop-up dialog.
```

![Screenshot that shows the "Connect to Active Directory Forest" page in the "Properties" window.](./media/how-to-connect-sync-change-serviceacct-pass/key6.png)

  
  

> #### 5. Reinitialize the password of the Entra ID Connector account

Reviewer: **BLOCKING — Product naming**: "Entra ID Connector account" must be "Microsoft Entra ID Connector account".

```suggestion
#### 5. Reinitialize the password of the Microsoft Entra ID Connector account
```

You can't directly provide the password of the Microsoft Entra ID connector account to the Synchronization Service. Instead, you need to use the cmdlet **Add-ADSyncAADServiceAccount** to reinitialize the Microsoft Entra ID Connector account. The cmdlet resets the account password and makes it available to the Synchronization Service:

5.1. Sign in to the Microsoft Entra Connect Sync server and open PowerShell.

> 5.2. To provide the Microsoft Entra ID Global Administrator credentials, run `$credential = Get-Credential`.

Reviewer: **Suggestion — Role naming**: The standard role name is "Global Administrator". "Microsoft Entra ID Global Administrator" isn't an official role name.

```suggestion
1. To provide Global Administrator credentials, run `$credential = Get-Credential`.
```

5.3. Run the cmdlet `Add-ADSyncAADServiceAccount -AADCredential $credential`.
 
   If the cmdlet is successful, the PowerShell command prompt appears. 
   

> The cmdlet resets the password for the Entra ID Connector account and updates it in Microsoft Entra ID and the Synchronization Service.

Reviewer: **BLOCKING — Product naming**: "Entra ID Connector account" must include "Microsoft".

```suggestion
The cmdlet resets the password for the Microsoft Entra ID Connector account and updates it in Microsoft Entra ID and the Synchronization Service.
```
